torch/ao/quantization/pt2e/duplicate_dq_pass.py

```diff
@@ -0,0 +1,73 @@
+import logging
+
+import torch
+from torch._export.pass_base import _ExportPassBase
+
+from torch.ao.quantization.pt2e.utils import filter_sym_size_users, is_valid_annotation
+
+from torch.fx.node import map_arg
+from torch.fx.passes.infra.pass_base import PassResult
+
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.WARNING)
+
+__all__ = ["DuplicateDQPass"]
+
+_DEQUANTIZE_OPS = [
+    torch.ops.quantized_decomposed.dequantize_per_tensor.default,
+    torch.ops.quantized_decomposed.dequantize_per_tensor.tensor,
+    torch.ops.quantized_decomposed.dequantize_per_channel.default,
+]
+
+
+"""
+This API is specifically to replace a specific use of producer node
+ATM, it uses internal API __update_args_kwargs, but ideally we mvoe
+this API to fx.Node.replace_use_with...
+"""
+
+
+def _copy_and_replace_use(
+    gm: torch.fx.GraphModule, producer: torch.fx.Node, user: torch.fx.Node
+):
+    with gm.graph.inserting_after(producer):
+        new_node = gm.graph.node_copy(producer)
+
+        def maybe_replace_node(n: torch.fx.Node) -> torch.fx.Node:
+            if n == producer:
+                return new_node
+            else:
+                return n
+
+        new_args = map_arg(user.args, maybe_replace_node)
+        new_kwargs = map_arg(user.kwargs, maybe_replace_node)
+        user.args = new_args
+        user.kwargs = new_kwargs
+
+
+def _maybe_duplicate_dq(
+    gm: torch.fx.GraphModule, producer: torch.fx.Node, user: torch.fx.Node
+):
+    annotation = user.meta.get("quantization_annotation", None)
+    if not is_valid_annotation(annotation):
+        return
+    _copy_and_replace_use(gm, producer, user)
+
+
+class DuplicateDQPass(_ExportPassBase):
+    """
+    kk
+    """
+
+    def call(self, graph_module: torch.fx.GraphModule) -> PassResult:
+        for node in graph_module.graph.nodes:
+            if node.op == "call_function" and node.target in _DEQUANTIZE_OPS:
+                dq_users = filter_sym_size_users(node)
```

Comment:
is this needed? can we just replace all users? also after the IR changes, we don't have sym size nodes anymore right?

Reply:
well I dont know if the sym size node will show up due to user code. like querying size on a tensor. I think here we are just making a call of not duplicating dq for sym size nodes. Either choice is fine. I would say we can go with this for now and change the behavior later